drivers/at_cmd/at_cmd.c

```diff
@@ -26,13 +27,20 @@ static K_THREAD_STACK_DEFINE(socket_thread_stack, \
 				CONFIG_AT_CMD_THREAD_STACK_SIZE);
 
 static K_SEM_DEFINE(cmd_pending, 1, 1);
+static K_MUTEX_DEFINE(list_mtx);
```

Comment:
Why do you use separate mutex for the notification handler slist? Doesn't that mean that the application can end up modifying the slist when it shouldn't?

Reply:
Ok, I see that this has be discussed before